### PR TITLE
Schematics model level validation

### LINKED ISSUE #106: Unable to add model-level validation with schematics
[schematics](https://schematics.readthedocs.io/en/latest/index.html) allows you to Type-level validation (when you want to validate only the value of a specific field) or [Model-level validation](https://schematics.readthedocs.io/en/latest/usage/validation.html#model-level-validation) (if you want to validate one field against the content of other field).

Considering a model and a validation method like the following:
```
class MyModel(Model):
    sale_price = DecimalType(required=True)
    list_price = DecimalType(required=True)

    def validate_list_price(self, data, value):
        if data['sale_price'] > data['list_price']:
            raise ValidationError(
                'List price must be greater or equal to sale price.')
        return value
```

When I execute the spider, I see this error for each item that fails the validation:
```
Traceback (most recent call last):
  File "/tmp/crawler/twisted/internet/defer.py", line 654, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/tmp/crawler/spidermon/contrib/scrapy/pipelines.py", line 110, in process_item
    ok, errors = validator.validate(data)
  File "/tmp/crawler/spidermon/contrib/validation/validator.py", line 22, in validate
    return not self.has_errors, self.errors
  File "/tmp/crawler/spidermon/contrib/validation/validator.py", line 40, in errors
    for field_name, messages in self._errors.items()])
  File "/tmp/crawler/spidermon/contrib/validation/validator.py", line 40, in <listcomp>
    for field_name, messages in self._errors.items()])
  File "/tmp/crawler/spidermon/contrib/validation/translator.py", line 12, in translate_messages
    return [self.translate_message(m) for m in messages]
  File "/tmp/crawler/spidermon/contrib/validation/translator.py", line 12, in <listcomp>
    return [self.translate_message(m) for m in messages]
  File "/tmp/crawler/spidermon/contrib/validation/translator.py", line 16, in translate_message
    if pattern.search(message):
TypeError: expected string or bytes-like object
```

### PR DESCRIPTION
**Pull request by rennerocha**
Fixes #106

### CODE REVIEW

File: spidermon/contrib/validation/translator.py [hunk 1]
```diff
@@ -2,6 +2,15 @@
...
 
+schematics_installed = False
+try:
+    import schematics
+
+    schematics_installed = True
+except ImportError:
+    pass
+
+
```
Comment (muzaffaryousaf) [inline spidermon/contrib/validation/translator.py]: This seems to be generic file for validations, can we not move schematics related logic inside schematics/translator ?

File: spidermon/contrib/validation/translator.py [hunk 2]
```diff
@@ -12,6 +21,14 @@ def translate_messages(self, messages):
         return [self.translate_message(m) for m in messages]
 
     def translate_message(self, message):
+        if (
+            schematics_installed
+            and message
+            and type(message) is schematics.datastructures.FrozenList
+        ):
+            # Necessary to handle model-level custom validators correctly
+            message = message[0].summary
```
Comment (muzaffaryousaf) [inline spidermon/contrib/validation/translator.py]: It would be nice if we can move schematic related logic in it's own file & class.

**Comment by raphapassini** [comment]
LGTM

File: spidermon/contrib/validation/translator.py (hunk 1 above)
Comment (rennerocha) [reply spidermon/contrib/validation/translator.py]: You are right. I am changing it.

**Comment by rennerocha** [comment]
@muzaffaryousaf I updated the code. Now schematics related code is isolated. Could you please review again?

**Pull request merged**